app/helpers/application_helper.rb

```diff
@@ -1,2 +1,6 @@
 module ApplicationHelper
+  def full_title page_title = ""
+    base_title = "Ruby on Rails Tutorial Sample App"
+    page_title.empty? ? base_title : "#{page_title} | #{base_title}"
```

Comment:
phân biệt nil?, empty?, blank?, present? giúp a

Reply:
+ nil? luôn trả về false trừ nil.nil?
+ empty là function của array, string, hash nên chỉ kiểu dữ liệu này mới có thể dùng được hàm này. empty? trả về true khi + string.length == 0, array không có phần tử, hash không có phần tử
+ blank? trả về true khi: nil, false, string độ dài bằng 0, string toàn khoảng trắng, array không có phần tử, hash không có phần tử
 + present? giá trị trả về ngược với blank?